appsec/src/extension/commands_helpers.c

```diff
@@ -42,16 +40,14 @@ typedef struct _dd_imsg {
     mpack_node_t root;
 } dd_imsg;
 
-// iif these two return success, _imsg_destroy must be called
-static inline dd_result _imsg_recv(
-    dd_imsg *nonnull imsg, dd_conn *nonnull conn);
-static inline ATTR_WARN_UNUSED dd_result _imsg_recv_cred(
+// iif this returns success, _imsg_destroy must be called
```

Comment:
typo

Reply:
no, it's correct, `_imsg_destroy` should be called if and only if `_imsg_recv` returns success. I did spot a branch where it was not being called, though, so I added a new commit to use a gcc/clang extension to prevent this.